lib/bolt/logger/util.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Bolt
+  module Logger
+    module Util
+      # This should be fine to track once per logger instance
+      def warn_once(msg)
```

Comment:
We should only warn once for a particular _kind_ of warning, not just the exact message. Like if multiple targets have the same config problem (maybe because you set it in the inventory), you should only see the warning one time. So we should have some sort of identifier here for "which" warning we're issuing (or we could take a formatter string and args to be interpolated).

On the same note, I'm unclear how having one `@warnings` per logger instance works with that. Will that cause us to still issue a warning once per target?

Reply:
For this specifically, it will still only warn once ever - it would make it so we couldn't use warn_once in a transport connection class, but I didn't see any particular warnings there currently that we'd want to warn_once and this makes managing the lifecycle of the warn_once *much* easier. I think the best alternative is to have a warn_once singleton - would that be better?